src/jobservice/job/impl/gc/util.go

```diff
@@ -62,6 +62,10 @@ func v2DeleteManifest(logger logger.Interface, repository, digest string) error
 		if err == readonly.Err {
 			return retry.Abort(err)
 		}
+		// If delete returned an err because the manifest is unknown, consider success
```

Comment:
the 404 is handled at https://github.com/goharbor/harbor/blob/main/src/jobservice/job/impl/gc/garbage_collection.go#L287

Reply:
it does, using a head request ince before entering the retry loop, however as I decribe in my comment here: https://github.com/goharbor/harbor/issues/18381#issuecomment-1477840214

you can see there is an edge case where the initial HEAD request returns 200, it enters the retry loop and tries to delete and gets a 500 at first and then on a subsequent retry gets a 404 which is treated as a retryable error (without the conditional I'm adding here) and eventually causes v2ManufesrDelete to return a not nil err.

this addresses this edge case.